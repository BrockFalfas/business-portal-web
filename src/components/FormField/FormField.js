import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Field } from 'formik'

import { Input } from 'antd'

export class FormField extends Component {
  static propTypes = {
    name: PropTypes.string.isRequired,
    component: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  }

  static defaultProps = {
    component: Input,
  }

  renderInput = ({ field, form }) => {
    const { component: Component, name, ...inputProps } = this.props

    const errors = form.touched[name] && form.errors[name]
    const helperText = inputProps.helperText || ''

    return (
      <Component
        {...field}
        {...inputProps}
        error={!!errors}
        helperText={errors || helperText}
      />
    )
  }

  render() {
    const { name } = this.props

    return <Field name={name} render={this.renderInput} />
  }
}

export default FormField
