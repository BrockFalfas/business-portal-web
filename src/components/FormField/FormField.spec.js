import React from 'react';
import { shallow } from 'enzyme';

import { FormField } from './FormField';

const initFormField = overrides => {
  const mockProps = {
    name: 'testName',
  };
  const wrapper = shallow(<FormField {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('component: <FormField/>', () => {
  it('should render without crashing', () => {
    const wrapper = initFormField();
    expect(wrapper).toMatchSnapshot();
  });
});
