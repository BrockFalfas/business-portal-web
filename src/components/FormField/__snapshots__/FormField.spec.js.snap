// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <FormField/> should render without crashing 1`] = `
Object {
  "mockProps": Object {
    "name": "testName",
  },
  "wrapper": <C
    name="testName"
    render={[Function]}
/>,
}
`;
